pkg/cmd/server/server.go

```diff
@@ -36,6 +42,25 @@ var ServerCmd = &cobra.Command{
 				return nil
 			}
 		}
+		p := RequirementProvider{}
+
+		requirements, err := p.Provider.CheckRequirements()
```

Comment:
The requirement check should be preformed in the server start command after the providers are registered. (https://github.com/daytonaio/daytona/blob/main/pkg/server/server.go#L120).

Also, if some of the requirements are not met, we should just log the warnings with a message `Provider requirement not met. The provider might not function correctly.`.
We shouldn't exit the process because the user might not want to use the provider or might want to delete it.

Reply:
> Also, if some of the requirements are not met, we should just log the warnings with a message `Provider requirement not met. The provider might not function correctly.`. We shouldn't exit the process because the user might not want to use the provider or might want to delete it.


for this do you mean this 

```go
for _, req := range *requirements {

    if req.Met {
    log.WithField("requirement", req.Reason).Info("Provider requirement met")
    } else {
    allRequirementsMet = false
    log.WithField("requirement", req.Reason).Warn("Provider requirement not met")
    }
    }
    
    if !allRequirementsMet {
    return  log.Warn("daytona server startup aborted, provider requirements not met")
    }
```

Using docker fr example, when requirements are met it will display this 

`INFO[0000] Requirement met: Docker is installed`
`INFO[0000] Requirement met: Docker is running`


When a requirement isn't met if docker is installed and not running it returns this 
```
INFO[0000] Requirement met: Docker is installed
WARN[0000] Requirement not met: Docker is not running. Error: [error message here]
```


When docker isn't installed 
`WARN[0000] Requirement not met: Docker is not installed`

Are these formats okay, for the other providers i returned an empty array, so their respective checkRequirements() doesn't get initialized. 
It just docker that performs the requirement checks

Reply:
> WARN[0000] Requirement not met: Docker is not running. Error: [error message here]

This is okay. Lets just put `Provider requirement not met....`